EngineLayer/GlycoSearch/GlycanBox.cs

```diff
@@ -103,7 +103,7 @@ public static IEnumerable<GlycanBox> BuildChildOGlycanBoxes(int maxNum, int[] gl
 
         public GlycanBox(int[] ids, bool targetDecoy = true):base(ids)
         {
-            byte[] kind = new byte[] { 0, 0, 0, 0, 0, 0, 0, 0, 0, 0 };
+            byte[] kind = new byte[Glycan.SugarLength];
```

Comment:
is length the number of sugars in the glycan, ignoring branching? maybe explain that with a comment somewhere